From b0788ec33a8c03d151bc3fdc11f4615fce7fff65 Mon Sep 17 00:00:00 2001
From: Peter Andreas Entschev <peter@entschev.com>
Date: Fri, 1 Nov 2019 06:38:30 -0700
Subject: [PATCH] UCM/CUDA: fix ucm_cudafree_dispatch_events assertion

Only call assert in ucm_cudafree_dispatch_events if cuMemGetAddressRange() call
succeeds and downgrade warning to debug message when it fails.
---
 src/ucm/cuda/cudamem.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/ucm/cuda/cudamem.c b/src/ucm/cuda/cudamem.c
index 078ded45b1..ad127dbda5 100644
--- a/src/ucm/cuda/cudamem.c
+++ b/src/ucm/cuda/cudamem.c
@@ -110,11 +110,12 @@ static void ucm_cudafree_dispatch_events(void *dptr)
     }
 
     ret = cuMemGetAddressRange(&pbase, &psize, (CUdeviceptr) dptr);
-    if (ret != CUDA_SUCCESS) {
-        ucm_warn("cuMemGetAddressRange(devPtr=%p) failed", (void *)dptr);
+    if (ret == CUDA_SUCCESS) {
+        ucs_assert(dptr == (void *)pbase);
+    } else {
+        ucm_debug("cuMemGetAddressRange(devPtr=%p) failed", (void *)dptr);
         psize = 1; /* set minimum length */
     }
-    ucs_assert(dptr == (void *)pbase);
 
     ucm_dispatch_mem_type_free((void *)dptr, psize, UCS_MEMORY_TYPE_CUDA);
 }
